docs/features/nullable-reference-types.md

```diff
@@ -157,9 +165,39 @@ var y = (IEnumerable<object>)x;  // warning
 var z = (IEnumerable<object?>)x; // no warning
 ```
 
+### Method type inference
+Method type inference uses the Conversions rules above to calculate top-level and nested nullability, in two attempts.
+The first attempt considers ~ distinct from ? and !. If that fails, a second attempt considers ~ implicitly convertible to ? and !.
+If the both attempts fail, a warning is reported that nullability could not be inferred.
+If the first attempt succeeds, the inferred types include nullability, otherwise the inferred types have unknown (~) nullability.
+```c#
+interface I<T> { }
+interface IOut<out T> { }
+T F2<T>(T x, T y) { ... }
+T F3<T>(T x, T y, T z) { ... }
+
+I<T?> x1;
+I<T~> y1;
+I<T!> z1;
+F2(x1, y1) // I<T~>, warning
```

Comment:
I expected no warning here.
The scenario that produces a warning is `F2(x1, z1)`